smart-hadoop-support/smart-hadoop-client/src/main/java/org/apache/hadoop/hdfs/SmartInputStream.java

```diff
@@ -39,12 +43,29 @@
     return fileState.getFileType();
   }
 
+  private static String getHadoopVersion() {
+    String version = VersionInfo.getVersion();
+    String[] parts = version.split("\\.");
+    if (parts.length < 2) {
+      throw new RuntimeException("Illegal Hadoop Version: " + version + " (expected A.B.* format)");
+    }
+    Integer first = Integer.parseInt(parts[0]);
+    if (first == 0 || first == 1) {
+      throw new RuntimeException("Hadoop version 0.x and 1.x are not supported");
+    }
+    Integer second = Integer.parseInt(parts[1]);
+    if (first == 2 && second <= 6) {
+      return HADOOP_26_VERSION;
+    } else {
+      return HADOOP_27_VERSION;
+    }
+  }
+
```

Comment:
This is a temp realization for Hadoop version check. Need refactor in a more decent way.